baselines/dom.asynciterable.generated.d.ts

```diff
@@ -2,14 +2,22 @@
 /// Window Async Iterable APIs
 /////////////////////////////
 
+interface FileSystemDirectoryHandleAsyncIterator<T> extends AsyncIteratorObject<T, BuiltinIteratorReturn, unknown> {
```

Comment:
I don't understand this, is FooAsyncIterator basically same as BarAsyncIterator, because I don't see a difference between this and ReadableStreamAsyncIterator below?

Reply:
They have independent prototypes on the web, so you could theoretically augment `ReadbleStreamAsyncIterator` independently via its prototype:

```js
var rs = new ReadableStream();
var proto = Object.getPrototypeOf(rs[Symbol.asyncIterator]());
proto; // [object ReadableStream AsyncIterator]
```

As with `ArrayIterator`, `MapIterator`, `SetIterator`, etc., we've opted to use interface names that more closely align with the underlying prototype.

Reply:
They have independent prototypes because the underlying iterator algorithms are different, but I wouldn't expect any independent members on them that would require separate declarations, at least not in the foreseeable future, because Web IDL simply doesn't have syntax for that and I don't see why IDL would want that.

@MattiasBuelens may have some opinion.

Reply:
Yes, but the general principle we've agreed on in https://github.com/microsoft/TypeScript/pull/58243 and in design meeting is to try to give these things a relevant name. `BuiltinIterator` was too broad and confusing, and `IteratorObject<T, BuiltinIteratorReturn, unknown>` was too unwieldy and verbose. Since we must give these names, we've opted to give them names that reflect the actual runtime reality.

Reply:
Hmmmm. I guess it would be fine as we don't have too many iterable interfaces here, although I still worry that this would encourage people do weird things that doesn't fit the spec intention. But matching with TS principle makes sense.